smartsim/ml/torch/data.py

```diff
@@ -26,77 +26,104 @@
 
 import numpy as np
 import torch
+from smartredis import Client, Dataset
 
-from smartsim.ml.data import DynamicDataDownloader, StaticDataDownloader
+from smartsim.ml.data import DataDownloader
 
 
-class StaticDataGenerator(StaticDataDownloader, torch.utils.data.IterableDataset):
-    """A class to download a dataset from the DB.
-
-    Details about parameters and features of this class can be found
-    in the documentation of ``StaticDataDownloader``, of which it is just
-    a PyTorch-specialized sub-class.
+class _TorchDataGenerationCommon(DataDownloader, torch.utils.data.IterableDataset):
+    def __init__(self, **kwargs):
+        init_samples = kwargs.pop("init_samples", False)
+        kwargs["init_samples"] = False
+        super().__init__(**kwargs)
+        if init_samples:
+            self.log(
+                "PyTorch DataLoader has to be created with ``init_samples``=False. Setting it to False automatically."
```

Comment:
@MattToast do you know if there is a standard for such things?

Reply:
Let's remove them for the moment being.

Reply:
@al-rigazzi not to my knowledge... if there is one, it might actually be ` ``init_samples=Fasle`` `.

From larger python ecosystem point of view, there may be an argument to follow the Zen of Python's "errors should never pass silently" statement and raise an exception if instead of trying to coerce the kwarg to a "valid" value via something like:
```py
def __init__(self, **kwargs):
    if "init_samples" in kwargs:
        raise TypeError(f"{type(self).__name__}.__init__ "
                         "got an unexpected keyword argument"
                         "'init_samples'")
    ...
    super().__init__(..., init_samples=False)
    ...
```
which would behave very similarly to if we explicitly wrote out all of the kwargs but left off `init_samples` so users didn't even have the opportunity to set it to a "wrong" value.
```py
def __init__(
    self,
    data_info_or_list,
    *,
    batch_size=32,
    ..., # etc
    # very explicitly leaving off `init_samples`
):
    super().__init__(data_info_or_list, batch_size=batch_size, ..., init_samples=False)
```
but of course there are always exceptions to this rule (pydantic constructors try to do bunch of type conversions for example), so really its up to basically up the author's discretion.

Long answer short, not really, lol ¯\\\_(ツ)\_/¯